File: src/main/java/domain/Name.java
package domain;

import static Message.ErrorMessage.CAR_NAME_LENGTH_ERROR_MESSAGE;

public class Name {

    public static final int MAX_NAME_LENGTH = 5;

    private final String value;

    public Name(String value) {
        validateNullAndLength(value);
        this.value = value;
    }

    private void validateNullAndLength(String value) {
        if (value == null || value.length() > MAX_NAME_LENGTH) {
            throw new IllegalArgumentException(CAR_NAME_LENGTH_ERROR_MESSAGE.getValue());
        }
    }

    public String getValue() {
        return value;
    }
}

File: src/main/kotlin/domain/Name.kt
package domain

import message.ErrorMessage

class Name(val value: String) {

    init {
        validateNullAndLength(value)
    }

    private fun validateNullAndLength(value: String) {
        require(value.length <= MAX_NAME_LENGTH) { ErrorMessage.CAR_NAME_LENGTH_ERROR_MESSAGE.value }
    }

    companion object {
        const val MAX_NAME_LENGTH = 5
    }
}
